pennylane/qnode.py

```diff
@@ -283,14 +292,23 @@ def construct(self, args, **kwargs):
             temp = [Variable(idx, name=key) for idx, _ in enumerate(_flatten(val))]
             kwarg_variables[key] = unflatten(temp, val)
```

Comment:
can we just skip this process if `cache=False`?

Reply:
Good point

Reply:
I placed this inside a `if self.cache:` block.